### PR TITLE
[TLS-Config] Enable TLS configuration for routes

### PR DESCRIPTION
**Pull request by varshaprasad96**
This PR enables the creation of secure https connection if
the path to TLS-Certificate and TLS-Key are provided by the
user. The connection falls back to non-https, if the required
parameters are not provided.

### CODE REVIEW

**Comment by varshaprasad96** [comment]
@awgreene @fahlmant - Please have a look at this PR. This enables a creation of secure https connection for the user.

**Comment by fahlmant** [comment]
First read over, looks good. Going to do some testing now. Thanks for the awesome work @varshaprasad96!

File: pkg/metrics/builder.go [hunk 1]
```diff
@@ -7,6 +7,8 @@ const (
 	defaultMetricsPath = "/customMetrics"
 	defaultMetricsPort = "8089"
 	defaultServiceName = "operator-metrics"
+	defaulttlsCertPath = ""
```
Comment (fahlmant) [inline pkg/metrics/builder.go:10]: Minor nit, can we change both of these to `defaultTLS`

**Comment by fahlmant** [comment]
Closing due to inactivity
@varshaprasad96 Please feel free to reopen!

**Pull request closed without merging**